src/components/BookList.js

```diff
@@ -0,0 +1,38 @@
+import React from "react";
+import PropTypes from "prop-types";
+import Book from "./Book";
+
+class BookList extends React.Component {
+  static propTypes = {
+    books: PropTypes.array.isRequired,
+    filters: PropTypes.object.isRequired
+  };
+
+  applyFilters = books => {
+    const filters = this.props.filters;
+    for (let filter in filters) {
+      if (filters[filter]) {
+        books = books.filter(book => book[filter]);
+      }
+    }
+    return books;
```

Comment:
This should be using `setState` so that rendering happens automatically. 

Actually, now that I'm looking at it, most of the variables in this class should be converted from props to state I think.

Reply:
https://reactjs.org/docs/thinking-in-react.html#step-4-identify-where-your-state-should-live
https://medium.com/netscape/component-state-vs-redux-store-1eb0c929277
https://lucybain.com/blog/2016/react-state-vs-pros/